### ISSUE TITLE
Nullability in generic method override

### ISSUE THREAD

**Issue by weifenluo** [dotnet/roslyn#53713]
**Version Used**:  C# 9.0, VS2019 Community Edition 16.9.5

**Steps to Reproduce**:

The following code generates CS0508/CS0453 in class `Test`:
```
using System.Diagnostics.CodeAnalysis;

public class Field<T>
{
    public T? Value { get; set; }
}

public abstract class Base
{
    public abstract T? GetValue<T>(Field<T> field);

    public abstract void SetValue<T>(Field<T> field, T? value);
}

public class Test : Base
{
    /*
    public override T? GetValue<T>(Field<T> field)  // This generates CS0508 and CS0453
    */
    [return: MaybeNull]
    public override T GetValue<T>(Field<T> field)
    {
        return field.Value;
    }

    /*
    public override void SetValue<T>(Field<T> field, T? value)  // This generates CS0453
    */
    public override void SetValue<T>(Field<T> field, [AllowNull]T value)
    {
        field.Value = value;
    }
}
```
**Expected Behavior**:
The generic method overrides:
```
public override T? GetValue<T>(Field<T> field)
```
and
```
public override void SetValue<T>(Field<T> field, T? value)
```
should not generate any compile error/warning because it has exactly same signature as its abstract declaration.

**Actual Behavior**:
The generic method overrides generates CS0508 and CS0453.

**Comment by cston** [comment]
@weifenluo, thanks for reporting this.

The errors reported for the overrides are expected since the compiler treats `T?` in an overridden method as `Nullable<T>` for compatibility with code written before C#8. To treat `T` as unconstrained in the override, add `where T : default` (see [sharplab.io](https://sharplab.io/#v2:D4AQDABCCMB0AiBLAhgcwHYHsDOAXRAxtrAMKYAmApgILrIA2AntotgNwCwAUNwMToBXevWQAjepQiU64yt24gAzFABMEAGKJK9cgB4AKgD5uAb24QLUZfoD8EAGoMBkkxFSVcbCNg9eAvtwBPFxKEGJ4AE7IBLiqEABCyD6m5pah4bhRMRC2EADiHo70zgaGABSa2npGEABmWjoAlJzBllZhopHRsSAALBAAyoVOlKUVDdWGdRMANDl2AG4jzYHyIcogavqUeBAAXAlJclxmXG2hmAuUERGIVPP5w8WjRuNVpdNVjRAA7gAW10k+n2ECotWQQlwqQspza5wA7J8dLAis4Wm0gtD2pdrrd7n1Bk8Sq9KjoPvUqnNcktnt9/oCciCwRD6FCzpZYXCLBTkajJABeCA0tFYoJ+IA===).

There is an open issue to improve the error message for these cases - see https://github.com/dotnet/roslyn/issues/46458.

**Comment by jcouv** [comment]
Closing as duplicate of https://github.com/dotnet/roslyn/issues/46458
